service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ActionCMOController.java

```diff
@@ -101,7 +108,15 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
 
         CaseManagementOrder orderWithDocument = actionCmoService.addDocument(order, document);
 
-        actionCmoService.prepareCaseDetailsForSubmission(caseDetails, orderWithDocument, hasJudgeApproved(order));
+        CaseManagementOrder orderWithNextHearing =
+            actionCmoService.appendNextHearingDateToCMO(caseData.getNextHearingDateList(), orderWithDocument);
+
+        actionCmoService.prepareCaseDetailsForSubmission(caseDetails, orderWithNextHearing, hasJudgeApproved(order));
+
+        String nextHearingDateLabel =
+            actionCmoService.createNextHearingDateLabel(orderWithNextHearing, caseData.getHearingDetails());
+
+        caseDetails.getData().put("nextHearingDateLabel", nextHearingDateLabel);
```

Comment:
would it worth moving lines 116 - 119 into _ prepareCaseDetailsForSubmission_?

Reply:
I'll make a note on this @Gogetter for now as I think that `prepareCaseDetailsForSubmission` needs to be renamed into a more specific name.